docs/madr/1-reassign-evolve.md

```diff
@@ -0,0 +1,105 @@
+# Reassign ANT to Evolve Implementation Kickoff on the ArNS Portal
+
+- Status: Proposed
+- Approvers: [Ariel], [Dylan], [Atticus], [Phil], [David]
+- Date: [2025-01-30]
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+Currently to evolve the ANT we use AOS `Eval` - this introduces the problem of
+evaluating what the current version of the code is. If we maintain a variable in
+the state of the process containing a version number, that tells us only what
+that version number is. At any point the user - owner - of the ANT may `Eval`
+new code into the process resulting in changes to the process that may not
+reflect what we expect the version number we assigned to reflect. To overcome
+this we dryrun interactions on APIs and analyze the results to see if the APIs
+perform as expected - this is very inefficient as dry run calls to CU's are
+expensive in networking client side, as well as for the CU.
+
+And so it is proposed to leverage the `Reassign-Name` handler in conjunction
```

Comment:
this is the output, not necessarily the contet/problem.

The main problem is upgrading ANTs doesn't guarantee users have the necessary APIs and specs to interact with the arns app. Additionally gateways expect ANTs to properly return names/comply with API specs and if they do not then arns name resolution is impacted

Reply:
[17f5a95](https://github.com/ar-io/ar-io-ant-process/pull/74/commits/17f5a959051231ca92bf0469ae591c9b9c25e887)